homework/hw08/task3.py

```diff
@@ -0,0 +1,69 @@
+# 3. Напишите функцию, которая получает на вход директорию и рекурсивно обходит её и все вложенные директории.
+# Результаты обхода сохраните в файлы json, csv и pickle.
+    # Для дочерних объектов указывайте родительскую директорию.
+    # Для каждого объекта укажите файл это или директория.
+    # Для файлов сохраните его размер в байтах,
+    # а для директорий размер файлов в ней с учётом всех вложенных файлов и директорий.
+# Соберите из созданных на уроке и в рамках домашнего задания функций пакет для работы с файлами разных форматов.
+
+import os
+import json
+import csv
+import pickle
+
+
+def get_size(path):
+    total_size = 0
+    for dir_path, dir_names, file_names in os.walk(path):
+        for file in file_names:
+            fp = os.path.join(dir_path, file)
+            total_size += os.path.getsize(fp)
+    return total_size
+
+
+def save_path_info(path, export_dir_name, export_file_name, data_type):
+    variable_data_type = ['json', 'csv', 'pickle']
+    data = []
+    for dir_path, dir_names, file_names in os.walk(path):
+        for file in file_names:
+            fp = os.path.join(dir_path, file)
+            data.append({
+                'Имя файла': file,
+                'Родительская директория': dir_path,
+                'Тип': 'file',
+                'Размер': os.path.getsize(fp)
+            })
+        for directory in dir_names:
+            dp = os.path.join(dir_path, directory)
+            data.append({
+                'Имя файла': directory,
+                'Родительская директория': dir_path,
+                'Тип': 'directory',
+                'Размер': os.path.getsize(dp)
+            })
+
+    if not os.path.exists(f'./{export_dir_name}'):
+        os.mkdir(export_dir_name)
+    export_file_name_full = f'./{export_dir_name}/{export_file_name}.{data_type}'
+
+    if data_type == 'json'.lower():
+        with open(export_file_name_full, 'w', encoding='utf-8') as f:
+            json.dump(data, f)
+    elif data_type == 'csv'.lower():
+        with open(export_file_name_full, 'w', encoding='utf-8') as f:
```

Comment:
Не забывайте указывать параметр `newline`, чтобы запись в csv происходила в каждой строке.
```suggestion
        with open(export_file_name_full, 'w', newline='', encoding='utf-8') as f:
```